### PR TITLE
Introduce negative numbers parsing

### LINKED ISSUE #328: shield: Correctly parse negative numbers
After #260 we'll be able to have negative numbers as result of math operations, but we aren't able to parse negative integer literals.

Fix the lexer to include the `-` sign in front of numbers in integers tokens. Add unit tests.

### PR DESCRIPTION
**Pull request by AzraelSec**
Hello, Warden Team 👋🏻 

I'd like to contribute to this project by opening a PR to make the shield DSL handle negative integers. I didn't comment on the related issue before picking it up since I briefly chatted with @Pitasi about this. Please forgive me if this was not the expected workflow to follow.

The current PR defines the abstraction to handle the negative sign as a prefix expression (with the highest priority between the operators) instead of changing the lexer to manually recognize negative integers. This way, you can write more complex expressions (e.g. `--1`, `-(-1)`, `-var` [*when/if constants/variables will be available*]) without additional costs.

Also, I used a small REPL tool to manually test the feature (as an additional step to write unit tests), and I decided to include it as part of the current PR. I'm happy to drop it if you don't think this might be useful.

I'm happy to discuss the proposed changes. Let me know if anything does not match what you expect. Thanks!

Closes #328 

## Summary by CodeRabbit

- **New Features**
  - Introduced a negative prefix operator to handle negative expressions and integers, enhancing arithmetic capabilities.

- **Bug Fixes**
  - Improved handling and evaluation of prefix expressions in various components.

- **Tests**
  - Added new test cases for evaluating boolean and arithmetic expressions involving negative numbers.
  - Added tests for parsing negative integers.

- **Documentation**
  - Updated documentation to reflect changes in handling prefix expressions.

- **Refactor**
  - Enhanced internal processing of prefix expressions for better performance and accuracy.

### CODE REVIEW

File: shield/internal/parser/parser_test.go [hunk 1]
```diff
@@ -100,6 +100,19 @@ func TestBooleanOperations(t *testing.T) {
...
+
+	expression := p.Parse()
+	require.NotNil(t, expression)
+
+	negative, ok := ast.UnwrapPrefixExpression(expression)
+	require.True(t, ok, "expression is not *ast.PrefixExpression. got=%T", expression)
+	require.Equal(t, "-", negative.Operator)
+}
+
 func TestParser(t *testing.T) {
```
Comment (pitasi) [inline shield/internal/parser/parser_test.go:116]: can we add some tests here to ensure that the priority of operators is the expected one?

**Review by pitasi — CHANGES_REQUESTED** [review]
Looking good!

I'll ask you to do a couple more little things but all the work looks good:

- add support for PrefixExpression to `stringify.go`
- add more test cases to `TestParses` (see comment) so we can see that priority among operators (InfixExpression, CallExperssion) is correct and stays like that

**Review by pitasi — APPROVED** [review]
thanks a lot ❤️

**Comment by AzraelSec** [comment]
It's been a pleasure to contribute! 😄

**Pull request merged**